Documentation/design-docs/COM-activation.md

```diff
@@ -0,0 +1,213 @@
+
+##### Registration-Free
+
+[RegFree COM for .NET](https://docs.microsoft.com/en-us/dotnet/framework/interop/configure-net-framework-based-com-components-for-reg) is another style of registration that does not require accessing the registry. This approach is complicated by the use of [application manifests](https://docs.microsoft.com/en-us/windows/desktop/SbsCs/application-manifests), but does have benefits for limiting environment impact and simplifying deployment. A severe limitation of this approach is that in order to use RegFree COM with a .NET class, the Window OS assumes the use of `mscoree.dll` for the in-proc server. Without a change in the Windows OS, this assumption in the RegFree .NET scenario makes usage of the current manifest approach a broken scenario for .NET Core.
+
+An example of a RegFree manifest for a .NET Framework class is below - note the absence of specifying a hosting server library (i.e. `mscoree.dll` is implied for the `clrClass` element).
+
+``` xml
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+    <assemblyIdentity
+        type="win32"
+        name="NetServer"
+        version="1.0.0.0">
+    </assemblyIdentity>
+    <clrClass
+        clsid="{3C58BBC9-3966-4B58-8EE2-398CBBC9FDC4}"
+        name="NetServer.Server"
+        threadingModel="Both"
+        runtimeVersion="v4.0.30319">
+    </clrClass>
+</assembly>
+```
+
+Due to the above issues with traditional RegFree manifests and .NET classes, an alternative system must be employed to enable a low-impact style of class registration for .NET Core.
+
+The proposed alternative for RegFree is as follows:
+
+1) The native application will define an application manifest, but instead of defining a `clrClass` node, the application will define at least 2 dependent assemblies. The first will be the host shim library (e.g. `CoreShim.dll`) that takes the place of `mscoree.dll` in .NET Framework. The second and any subsequent assemblies will be those managed assemblies that contain the managed class(s) to activate.
+    ``` xml
+    <?xml version="1.0" encoding="utf-8" standalone="yes" ?>
+    <assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+        <assemblyIdentity
+            type="win32"
+            name="COMClientPrimitives"
+            version="1.0.0.0"/>
+
+        <dependency>
+            <dependentAssembly>
+            <!-- RegFree COM - CoreCLR Shim -->
+            <assemblyIdentity
+                type="win32"
+                name="CoreShim.X"
+                version="1.0.0.0"/>
+            </dependentAssembly>
+        </dependency>
+
+        <dependency>
+            <dependentAssembly>
+            <!-- RegFree COM - Managed assembly -->
+            <assemblyIdentity
+                type="win32"
+                name="NETServer"
```

Comment:
I'd be careful here on naming, We should be clear this is a COM server, not to be confused with the ASP.NET Web Server.

Reply:
That is purely an example for illustrative purposes.